eslintPlugin/src/noAnimatedStyleToNonAnimatedComponent.ts

```diff
@@ -0,0 +1,175 @@
+import { AST_NODE_TYPES, TSESLint, TSESTree } from '@typescript-eslint/utils';
+
+const rule: TSESLint.RuleModule<'animatedStyle' | 'sharedValue'> = {
+  create: function (context) {
+    return {
+      JSXOpeningElement(node: TSESTree.JSXOpeningElement) {
+        if (node.name.type === AST_NODE_TYPES.JSXMemberExpression) return; // Property-like namespace syntax <Animated.View>
+        if (node.name.type === AST_NODE_TYPES.JSXNamespacedName) return; // XML-based namespace syntax: <Animated:View>
+
+        const sourceCode = context.getSourceCode();
+        const tokensBefore = sourceCode.getTokensBefore(node);
+        const componentName = node?.name?.name;
+
+        main();
+        function main() {
+          if (
+            isVariableDefinedAs(componentName, 'Animated') || //TODO allow to customize rule and provide custom import name
+            isVariableDefinedAs(componentName, 'Reanimated') ||
+            isVariableDefinedAs(componentName, 'createAnimatedComponent')
+          ) {
+            return;
+          }
+          const styleAttribute = node.attributes
+            .map((attribute) => {
+              return attribute.type === AST_NODE_TYPES.JSXAttribute &&
+                attribute.name.name === 'style'
+                ? [attribute]
+                : [];
+            })
+            .flat();
+
+          if (styleAttribute.length == 0) {
+            return;
+          }
+
+          const styleValue = styleAttribute[0].value; // assume no duplicate props
+          if (
+            styleValue === null ||
+            styleValue.type === AST_NODE_TYPES.Literal
+          ) {
+            return; //incorrect styles
+          }
+
+          if (styleValue.type === AST_NODE_TYPES.JSXSpreadChild) {
+            return; //TODO Ignore this for now
+          }
+
+          const styleExpression = styleValue.expression;
+          switch (styleExpression.type) {
+            case AST_NODE_TYPES.Identifier: // style={myVariable}
+              checkIdentifierNodeForBeingAnimated(styleExpression);
+              break;
+            case AST_NODE_TYPES.ArrayExpression: // style={[style1, style2]}
+              checkArrayNodeForBeingAnimated(styleExpression);
+              break;
+            case AST_NODE_TYPES.ObjectExpression: //style={{backgroundColor:'pink'}}
+              checkObjectNodeForBeingAnimated(styleExpression);
+              break;
+            case AST_NODE_TYPES.MemberExpression: //style={{backgroundColor:styles.myStyle}}
+              //We assume that all member expressions are correct
+              break;
+          }
```

Comment:
How about moving it to a separate function?

Reply:
I am not really convinced if its worth moving. We could split the whole `main` function into smaller functions instead.